BraveRewardsUI/Wallet/WalletViewController.swift

```diff
@@ -317,23 +319,37 @@ class WalletViewController: UIViewController, RewardsSummaryProtocol {
   }
   
   @objc private func tappedMonthlyTip() {
-    // TODO: Replace with actual values.
-    let options = [
-      BATValue(1),
-      BATValue(5),
-      BATValue(10)
-    ]
-    let optionsVC = OptionsSelectionViewController(options: options, selectedOptionIndex: 1, optionSelected: { [weak self] index in
-      // TODO: save selection and update UI
-      guard let self = self else {
-        return
+    guard let host = state.url.host else { return }
+    state.ledger.publisherBanner(forId: host, completion: { [weak self] banner in
+      guard let self = self else { return }
+      
+      var options = TippingViewController.defaultTippingAmounts.map({ BATValue($0) })
+      if let banner = banner, !banner.amounts.isEmpty {
+        options = banner.amounts.map({ BATValue($0.doubleValue) })
       }
-      self.navigationController?.popToViewController(self, animated: true)
-      // swiftlint:ignore:next
-      self.publisherSummaryView.monthlyTipView.batValueView.amountLabel.text = options[safe: index]?.displayString ?? options[0].displayString
+      
+      options.insert(BATValue(0.0), at: 0)
+      
+      let selectedIndex = options.firstIndex(where: { Int($0.doubleValue) == Int(self.recurringTipAmount) }) ?? 0
+      
+      let optionsVC = TippingSelectionViewController(state: self.state, options: options, selectedOptionIndex: selectedIndex, optionSelected: { [weak self] optionIndex in
+        guard let self = self else { return }
+        
+        self.recurringTipAmount = options[optionIndex].doubleValue
+        
+        self.navigationController?.popToViewController(self, animated: true)
+        // swiftlint:ignore:next
+        self.publisherSummaryView.monthlyTipView.batValueView.amountLabel.text = options[safe: optionIndex]?.displayString ?? options[0].displayString
+        
+        // The user decided to remove an existing tip.
+        if optionIndex == 0 && selectedIndex > 0 {
```

Comment:
Probably safer to check if the option (`BATValue`)'s value is 0 instead of just checking the option index. 

I don't think the second check to `selectedIndex` is needed either since choosing zero regardless of what you had previously selected should remove the tip, considering you can't get to this screen without having a non-zero tip

Reply:
Fixed.